lib/httpoison.ex

```diff
@@ -134,7 +134,7 @@ defmodule HTTPoison.Request do
     headers = request.headers |> Enum.map(fn {k, v} -> "-H '#{k}: #{v}'" end) |> Enum.join(" ")
 
     body =
-      case request.body do
+      case HTTPoison.Base.maybe_process_form(request.body) do
```

Comment:
I opted to make this a function in `HTTPoison.Base` so that we can use the same logic for generating curl args vs hackney args